promql/engine.go

```diff
@@ -651,57 +654,132 @@ func (ng *Engine) populateSeries(ctx context.Context, querier storage.Querier, s
 		warnings  storage.Warnings
 		err       error
 	)
+	RemoteSelectCalls := 0
+	if querier.Remotely() {
+		if err := ng.remoteReadGate.Start(ctx); err != nil {
+			return nil, err
+		}
+		parser.Inspect(s.Expr, func(node parser.Node, path []parser.Node) error {
+			switch node.(type) {
+			case *parser.VectorSelector:
+				RemoteSelectCalls++
+			}
+			return nil
+		})
+	}
+	if RemoteSelectCalls < 2 {
```

Comment:
Would it significantly affect performance to move that `if` inside the `switch` and  thus avoid code duplication?

Reply:
[outside]
![image](https://user-images.githubusercontent.com/9583245/78523649-0f4b0b00-7804-11ea-90fc-bfaffc12a7c6.png)
[inside]
![image](https://user-images.githubusercontent.com/9583245/78523717-41f50380-7804-11ea-9d5e-71b12291fc28.png)